examples/apple/coreml/scripts/inspector_utils.py

```diff
@@ -0,0 +1,429 @@
+# Copyright © 2024 Apple Inc. All rights reserved.
+#
+# Please refer to the license found in the LICENSE file in the root directory of the source tree.
+
+import copy
+import errno
+import json
+import os
+
+import subprocess
+from dataclasses import dataclass
+from pathlib import Path
+
+from typing import Any, Dict, Final, List, Optional, Tuple, Union
+
+import executorch.exir as exir
+
+import pandas as pd
+import torch
+from executorch.backends.apple.coreml.compiler import CoreMLBackend
+from executorch.backends.apple.coreml.partition import CoreMLPartitioner
+
+from executorch.exir import (
+    EdgeProgramManager,
+    ExecutorchBackendConfig,
+    ExecutorchProgramManager,
+    ExirExportedProgram,
+    to_edge,
+)
+from executorch.exir.backend.compile_spec_schema import CompileSpec
+from executorch.exir.tracer import Value
+
+from executorch.sdk import BundledProgram, generate_etrecord, Inspector
+
+from executorch.sdk.bundled_program.config import MethodTestCase, MethodTestSuite
+from executorch.sdk.bundled_program.serialize import (
+    serialize_from_bundled_program_to_flatbuffer,
+)
+from executorch.sdk.inspector import Event
+
+from torch.export import export, ExportedProgram
+
+COREML_METADATA_KEYS: Final[List[Tuple[str, str]]] = [
+    ("operatorName", "coreml_operator"),
+    ("estimatedCost", "coreml_estimated_cost"),
+    ("preferredComputeUnit", "coreml_preferred_device"),
+    ("supportedComputeUnits", "coreml_supported_devices"),
+]
+
+
+def build_sdk_runner_including_coreml(
+    root_dir_path: Path,
+    conda_env_name: str,
+    force: bool = False,
+):
+    if not force:
+        sdk_executable_path = (
+            root_dir_path / "cmake-out" / "examples" / "sdk" / "sdk_example_runner"
+        )
+        print(sdk_executable_path)
+        if sdk_executable_path.is_file():
+            return
+
+    cd_root_command: str = f"cd {root_dir_path.resolve()}"
+    conda_activate_env_command: str = f"source conda activate {conda_env_name}"
+    build_sdk_runner_command: str = (
+        "./examples/sdk/build_sdk_example_runner.sh --coreml"
+    )
+    build_command: str = (
+        f"{cd_root_command}; {conda_activate_env_command}; {build_sdk_runner_command}"
```

Comment:
shouldn't these be joined with `&&` so that we will terminate early instead of getting confused if a subcommand fails?

Reply:
Yes, thank you!